tests/test_launcher.py

```diff
@@ -722,51 +740,79 @@ def test_ip_and_start_instance(
         (ip_envvar is None or ip_envvar == "") and bool(ip)
     )
 
+    ###################
+    # Exception case: start_instance and ip are passed as args.
     exceptions = start_instance_envvar is None and start_instance is None and ip_is_true
-
     if (start_instance_is_true and ip_is_true) and not exceptions:
+        if not _HAS_ATP:
+            pytest.skip("Requires 'ansys-tools-path'. ")
+
         with pytest.raises(
             ValueError,
             match="When providing a value for the argument 'ip', the argument ",
         ):
             options = launch_mapdl(
-                start_instance=start_instance, ip=ip, _debug_no_launch=True
+                exec_file=find_ansys()[0],
+                start_instance=start_instance,
+                ip=ip,
+                _debug_no_launch=True,
             )
+        return  # Exit early the test
 
-        return  # Exit
+    ###################
+    # Faking MAPDL launching and returning args.
+    if not _HAS_ATP:
+        pytest.skip("Requires 'ansys-tools-path'. ")
 
     if (
         isinstance(start_instance_envvar, bool) and isinstance(start_instance, bool)
     ) or (ip_envvar and ip):
         with pytest.warns(UserWarning):
             options = launch_mapdl(
```

Comment:
this kind of change to tests make me nervous. To me it implies you broke compatiblity of `launch_mapdl`.  Are you sure you didn't?

Reply:
Very good point. I will double check.